lang/java/avro/src/main/java/org/apache/avro/io/JsonDecoder.java

```diff
@@ -453,12 +445,14 @@ public Symbol doAction(Symbol input, Symbol top) throws IOException {
       Symbol.FieldAdjustAction fa = (Symbol.FieldAdjustAction) top;
       String name = fa.fname;
       if (currentReorderBuffer != null) {
-        List<JsonElement> node = currentReorderBuffer.savedFields.get(name);
-        if (node != null) {
-          currentReorderBuffer.savedFields.remove(name);
-          currentReorderBuffer.origParser = in;
-          in = makeParser(node);
-          return null;
+        try (TokenBuffer tokenBuffer = currentReorderBuffer.savedFields.get(name)) {
```

Comment:
we could call currentReorderBuffer.savedFields.remove here instead and be more efficient

Reply:
That looks like a reasonable change, although I was aiming for the smallest diff in this section.